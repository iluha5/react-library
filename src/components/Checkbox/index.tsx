import * as React from 'react';
import { FC, InputHTMLAttributes } from 'react';

// import * as style from './style.scss';
const style = require('./style.scss');

interface IProps extends InputHTMLAttributes<HTMLInputElement>{
    id: string
}

const Checkbox: FC<IProps> =  ({id, ...rest}) => {
    return (
        <div className={style['Checkbox']} >
            <input
                type="checkbox"
                id={id}
                {...rest}
            />
            <label
                htmlFor={id}
            >
            </label>
        </div>
    );
};

export default Checkbox;
