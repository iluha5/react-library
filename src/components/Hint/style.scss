@import "../../variables";

.Hint {
    min-width: 250px;
    overflow: hidden;

    text-align: left;

    z-index: $zindex-third-level;
}

.Hint-Header {
    padding: 10px 21px 12px 21px;

    font-weight: 600;
    font-size: 14px;
    line-height: 19px;

    background: linear-gradient(91.17deg, rgba(218, 113, 99, 0.3) 1.75%, rgba(218, 113, 99, 0) 101.32%);
}

.Hint-Content {
    padding: 10px 21px 12px 21px;

    color: $text-second-color;
}
