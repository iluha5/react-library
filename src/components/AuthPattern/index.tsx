//@ts-nocheck

import * as React from 'react';

import cn from 'classnames';

import style from './style.scss';

import Paper from 'components/Paper';

interface IProps {
    image?: string,
    children: React.ReactNode
}

const AuthPattern: React.FC<IProps> = ({ children, image }) => {
    return (
        <div className={style['AuthPattern']}>
            <div className={cn(
                style['AuthPattern-Modal'],
                image && style['AuthPattern-Modal_WithImage']
            )}>
                <Paper
                    className={style['AuthPattern-Main']}
                    background={'rgba(255,255,255,0)'}
                >
                    <div className={style['AuthPattern-PaperContainer']}>
                        <div className={style['AuthPattern-Pattern']}> </div>
                        <div className={style['AuthPattern-ContentContainer']}>
                            <div className={style['AuthPattern-Content']}>
                                <div className={style['AuthPattern-ImageWrapper']}>
                                    {image ?
                                        <img className={style['AuthPattern-Logo']} src={image} alt=""/>
                                        :
                                        null
                                    }
                                    {/*<img className={style['AuthPattern-Logo']} src={logo} alt=""/>*/}
                                </div>

                                {children}
                            </div>
                        </div>

                    </div>
                </Paper>
            </div>
        </div>
    );
};

export default AuthPattern;
