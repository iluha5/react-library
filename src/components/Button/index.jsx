import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import style from './style.scss';

function Button ({btnType, className, size, children,  ...rest}) {
    return (
        <button
            className={cn(
                style.Btn,
                style[`Btn_size_${size}`],
                style[`Btn_type_${btnType}`],
                className,
            )}
            {...rest}
        >
            <span className={style['Btn-Label']}>{children}</span>
        </button>
    );
}

Button.defaultProps = {
    btnType: 'primary',
    size: 'm',
    disabled: false,
};

Button.propTypes = {
    className: PropTypes.any,
    btnType: PropTypes.oneOf(['primary', 'secondary', 'success', 'link', 'fake']),
    size: PropTypes.string,
    disabled: PropTypes.bool,
    children: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.object,
    ]),
    onClick: PropTypes.any,
};

export default Button;
