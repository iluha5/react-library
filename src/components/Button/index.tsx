import * as React from 'react';
import cn from 'classnames';

import style from './style.scss';

interface IProps extends React.HTMLProps<HTMLButtonElement> {
    btnType?: 'primary' | 'secondary' | 'success' | 'link' | 'fake',
    btnSize?: 'm' | 'xs',
    type?: 'button' | 'submit' | 'reset' | undefined
}

const Button: React.FC<IProps> = ({ btnType = 'primary', className, btnSize = 'm', children, ...rest }) => {
    return (
        <button
            className={cn(
                style.Btn,
                style[`Btn_size_${btnSize}`],
                style[`Btn_type_${btnType}`],
                className,
            )}
            {...rest}
        >
            <span className={style['Btn-Label']}>{children}</span>
        </button>
    );
};

export default Button;
