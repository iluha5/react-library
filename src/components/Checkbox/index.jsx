import React from 'react';
import PropTypes from 'prop-types';

import style from './style.scss';

function Checkbox({id, ...rest}) {
    return (
        <div className={style['Checkbox']} >
            <input
                type="checkbox"
                id={id}
                {...rest}
            />
            <label
                htmlFor={id}
            >
            </label>
        </div>
    );
}

Checkbox.propTypes = {
    id: PropTypes.string.isRequired,
};

export default Checkbox;
