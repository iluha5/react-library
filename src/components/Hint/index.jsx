import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import style from './style.scss';
import Paper from 'components/Paper';
import withAnimation from 'containers/WithAnimation';

export function Hint({ header, content, className }){
    return (
        <Paper
            className={cn(
                style['Hint'],
                className
            )}
            background={'rgb(247, 247, 249)'}
        >
            {header &&
                <div className={style['Hint-Header']}>
                    {header}
                </div>
            }
            {content &&
                <div className={style['Hint-Content']}>
                    {content}
                </div>
            }
        </Paper>
    );
}

Hint.propTypes = {
    header: PropTypes.string,
    content: PropTypes.string,
    className: PropTypes.string,
};

export default withAnimation(Hint);
