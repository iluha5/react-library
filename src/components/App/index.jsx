import React from 'react';
import PropTypes from 'prop-types';
import {withRouter} from 'react-router-dom';

import cn from 'classnames';

import {TIME_OVER, TEN_SECONDS_LEFT} from 'utils/constants';
import style from './style.scss';

import Button from 'components/Button';
import Checkbox from 'components/Checkbox';
import TimerContainer from 'containers/Timer';
import Modal from 'components/Modal';
import Input from 'components/Input';
import Icon from 'components/Icon';
import MultiEmailsExample from 'components/MultiEmailsExample';
import LoginContainer from 'containers/Login';

function App({
    handlerButtonClick,
    triggerBox1,
    isBox1Checked,
    timeLeft,
    handlerTime,
    triggerCheckboxButtonClick,
    timerID,
    isCheckboxModalOpen,
    handlerTimeButtonClick,
    isEmailErrors,
    onEmailChange,
    handlerLoginClick,
    isLoginModalOpen,
    history,
}) {
    const checkboxModalData = {
        header: 'Important message',
        content: 'I\'m popup message. Try to close me with mouse or keyboard. Bye!',
    };

    const loginModalData = {
        header: 'Login',
        content: <LoginContainer/>,
    };

    return (
        <div className={style['App']}>
            <h2 className={style['App-Title']}>
                Buttons and Notifications.
            </h2>
            <h3 className={style['App-SubTitle']}>
                Try to click them.
            </h3>
            <div className={style['App-Line']}>
                <div className={style['App-Item']}>
                    <Button
                        onClick={handlerButtonClick('Primary', true)}
                    >
                        Primary
                    </Button>
                </div>
                <div className={style['App-Item']}>
                    <Button
                        onClick={handlerButtonClick('Secondary', false)}
                        btnType={'secondary'}
                    >
                        Secondary
                    </Button>
                </div>
                <div className={style['App-Item']}>
                    <Button
                        onClick={handlerButtonClick('Success', true)}
                        btnType={'success'}
                    >
                        Success
                    </Button>
                </div>
                <div className={style['App-Item']}>
                    <Button
                        onClick={handlerButtonClick('Fake', false)}
                        btnType={'fake'}
                    >
                        Fake
                    </Button>
                </div>
                <div className={style['App-Item']}>
                    <Button
                        onClick={handlerButtonClick('Link', true)}
                        btnType={'link'}
                    >
                        Link
                    </Button>
                </div>
            </div>
            <h2 className={style['App-Title']}>
                Custom Checkboxes and Popups.
            </h2>
            <h3 className={style['App-SubTitle']}>
                Try it!
            </h3>
            <div className={style['App-Line']}>
                <div className={style['App-Item']}>
                    <Checkbox
                        id='box1'
                        onChange={triggerBox1}
                    />
                </div>
                <div className={style['App-Item']}>
                    <Button
                        onClick={triggerCheckboxButtonClick}
                        disabled={!isBox1Checked}
                    >
                        Open popup
                    </Button>
                </div>
            </div>
            <h2 className={style['App-Title']}>
                Custom Input with Animated Hints.
            </h2>
            <h3 className={style['App-SubTitle']}>
                Input some email.
            </h3>
            <div className={style['App-Line']}>
                <div className={cn(style['App-Item'], style['App-EmailWrapper'])}>
                    <Input
                        error={isEmailErrors}
                        className={style['App-EmailInput']}
                        type='email'
                        name='email'
                        placeholder='Type an email'
                        onChange={onEmailChange}
                        endIcon={isEmailErrors &&
                        <Icon
                            name={'warning'}
                            size={'sm'}
                        />
                        }
                        renderModalHint={isEmailErrors}
                        modalHintData={{
                            header: 'Enter an email',
                            content: 'Please, enter a valid email',
                        }}
                    />
                </div>
            </div>
            <h2 className={style['App-Title']}>
                WebWorker Timer.
            </h2>
            <h3 className={style['App-SubTitle']}>
                Click to start the timer.
            </h3>
            <div className={style['App-Line']}>
                <div className={cn(style['App-Item'], style['App-TimerWrapper'])}>
                    <TimerContainer
                        key={timerID}
                        timeLeft={timeLeft}
                        onTimeOver={timeLeft !== 0 ? handlerTime(TIME_OVER) : () => {
                        }}
                        trigger={{
                            time: 10,
                            callback: handlerTime(TEN_SECONDS_LEFT),
                        }}
                    />
                </div>
                <div className={style['App-Item']}>
                    <Button
                        onClick={handlerTimeButtonClick}
                        disabled={timeLeft !== 0}
                    >
                        Take 15 sec
                    </Button>
                </div>
            </div>
            <h2 className={style['App-Title']}>
                Multiple Emails Input.
            </h2>
            <h3 className={style['App-SubTitle']}>
                Open the input.
            </h3>
            <div className={style['App-Line']}>
                <div className={cn(style['App-Item'], style['App-MultiEmailsWrapper'])}>
                    <MultiEmailsExample/>
                </div>
            </div>
            <h2 className={style['App-Title']}>
                Login form.
            </h2>
            <h3 className={style['App-SubTitle']}>
                {/*Please login with <strong>test@test.test</strong> and <strong>11111111</strong>*/}
                Try to login.
            </h3>
            <div className={style['App-Line']}>
                <div className={style['App-Item']}>
                    <Button
                        // onClick={handlerLoginClick}
                        onClick={() => history.push('/login')}
                    >
                        Go to login form
                    </Button>
                </div>
            </div>

            {isCheckboxModalOpen &&
            <Modal
                header={checkboxModalData.header}
                content={checkboxModalData.content}
                handleClose={triggerCheckboxButtonClick}
            />
            }

            {isLoginModalOpen &&
            <Modal
                header={loginModalData.header}
                content={loginModalData.content}
                handleClose={handlerLoginClick}
            />
            }
        </div>
    );
}

App.propTypes = {
    handlerButtonClick: PropTypes.func.isRequired,
    triggerBox1: PropTypes.func.isRequired,
    isBox1Checked: PropTypes.bool.isRequired,
    isCheckboxModalOpen: PropTypes.bool.isRequired,
    isEmailErrors: PropTypes.bool.isRequired,
    timeLeft: PropTypes.number.isRequired,
    handlerTime: PropTypes.func.isRequired,
    onEmailChange: PropTypes.func.isRequired,
    handlerTimeButtonClick: PropTypes.func.isRequired,
    triggerCheckboxButtonClick: PropTypes.func.isRequired,
    timerID: PropTypes.number.isRequired,
    handlerLoginClick: PropTypes.func.isRequired,
    isLoginModalOpen: PropTypes.bool.isRequired,
    history: PropTypes.shape({
        push: PropTypes.func,
    }),
};

export default withRouter(App);
