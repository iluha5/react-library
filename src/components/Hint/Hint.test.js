import React from 'react';

import { Hint } from 'components/Hint';
import {shallow, mount} from 'enzyme';
// import {checkProps, findByClassName} from 'utils/testUtils';

describe('Hint component', () => {

    it('Render Button with props 1', () => {
        const props = {
            header: 'Test Hint',
            content: 'Test Content',
        };
        const hint = shallow(<Hint {...props} />);

        expect(hint).toMatchSnapshot();
    });

    it('Render Hint content block', () => {
        const props = {
            header: 'Test Hint',
            content: 'Test Content',
        };
        const wrap = mount(<Hint {...props} />);

        expect(wrap.find('.Hint-Content').text()).toBe('Test Content');
    });
});
