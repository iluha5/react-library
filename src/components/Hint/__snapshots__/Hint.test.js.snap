// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint component Render Button with props 1 1`] = `
<ForwardRef
  background="rgb(247, 247, 249)"
  className="Hint"
  depth={1}
>
  <div
    className="Hint-Header"
  >
    Test Hint
  </div>
  <div
    className="Hint-Content"
  >
    Test Content
  </div>
</ForwardRef>
`;
